lib/logstash/codecs/plain.rb

```diff
@@ -24,15 +31,27 @@ class LogStash::Codecs::Plain < LogStash::Codecs::Base
   # This only affects "plain" format logs since json is `UTF-8` already.
   config :charset, :validate => ::Encoding.name_list, :default => "UTF-8"
 
-  MESSAGE_FIELD = "message".freeze
+  def initialize(*params)
+    super
+
+    @original_field = ecs_select[disabled: nil, v1: '[event][original]']
 
-  def register
     @converter = LogStash::Util::Charset.new(@charset)
     @converter.logger = @logger
   end
 
+  def register
+    # no-op
+  end
+
+  MESSAGE_FIELD = "message".freeze
+
   def decode(data)
-    yield LogStash::Event.new(MESSAGE_FIELD => @converter.convert(data))
+    message = @converter.convert(data)
+    event = event_factory.new_event
+    event.set MESSAGE_FIELD, message
+    event.set @original_field, message.dup.freeze if @original_field
```

Comment:
using the charset converted field as `event.original`, same as for `message`,
but I wondered whether using the raw `data` would be more appropriate ...
(this use-case is common across codecs - we chose to set the converted value in json codec)